lib/rubygems/security.rb

```diff
@@ -490,9 +492,13 @@ def self.create_key(algorithm)
       when 'rsa'
         OpenSSL::PKey::RSA.new(RSA_DSA_KEY_LENGTH)
       when 'ec'
-        domain_key = OpenSSL::PKey::EC.new(EC_NAME)
-        domain_key.generate_key
-        domain_key
+        if RUBY_VERSION >= "2.4.0"
```

Comment:
Is this version check necessary these days? Wouldn't this branch be enough, because Ruby 2.4 is not supported anyway.

Reply:
It is still supported. We can drop it once we drop ruby 2.3 support, so the explicit check for the version is useful so that we don't forget to cleanup.